tests/ResourceSyncingTest.php

```diff
@@ -628,9 +880,148 @@ public function getCentralModelName(): string
+    public function getSyncedCreationAttributes(): array
+    {
+        // Default values when creating resources from tenant to central DB
+        return
+            [
+                'name' => 'Default Name',
+                'email' => 'default@localhost',
+                'password' => 'password',
+                'role' => 'admin',
+                'foo' => 'bar'
+            ];
+    }
+}
+
+// override method in ResourceUser class to return attribute names
+class ResourceUserProvidingAttributeNames extends ResourceUser {
+    public function getSyncedCreationAttributes(): array
+    {
+        // Attributes used when creating resources from tenant to central DB
+        // Notice here we are not adding "code" filed because it doesn't
+        // exist in central model
+        return
+            [
+                'name',
+                'password',
+                'email',
+                'role',
+                'foo' => 'bar'
+            ];
+    }
+
+}
+
+// override method in CentralUser class to return attribute default values
+class CentralUserProvidingDefaultValues extends CentralUser {
+    public function getSyncedCreationAttributes(): array
+    {
+        // Attributes default values when creating resources from central to tenant model
+        return
+            [
+                'name' => 'Default User',
+                'email' => 'default@localhost',
+                'password' => 'password',
+                'role' => 'admin',
+            ];
+    }
+}
+
+// override method in CentralUser class to return attribute names
+class CentralUserProvidingAttributeNames extends CentralUser {
+    public function getSyncedCreationAttributes(): array
+    {
+        // Attributes used when creating resources from central to tenant DB
+        return
+            [
+                'global_id',
+                'name',
+                'password',
+                'email',
+                'role',
+            ];
+    }
+}
+
+/*
+ * Below classes are sync resources setup classes for different schemas
+ *
+ * ResourceTenantForDifferentSchema is tenant model with different pivot table name
+ */
+class ResourceTenantForDifferentSchema extends Tenant
+{
+    public function users()
+    {
+        return $this->belongsToMany(CentralUserForDifferentSchema::class, 'tenant_central_users', 'tenant_id', 'global_user_id', 'id', 'global_id')
+            ->using(TenantPivot::class);
+    }
+}
+
+class CentralUserForDifferentSchema extends CentralUser {
+    public $table = 'central_users';
+
+    public function tenants(): BelongsToMany
+    {
+        return $this->belongsToMany(ResourceTenantForDifferentSchema::class, 'tenant_central_users', 'global_user_id', 'tenant_id', 'global_id')
+            ->using(TenantPivot::class);
+    }
+
+    public function getTenantModelName(): string
+    {
+        return ResourceUserForDifferentSchema::class;
+    }
+}
+
+class ResourceUserForDifferentSchema extends ResourceUser {
```

Comment:
You use the `different schema` terminology a lot here. Can you explain what you mean by it? The naming isn't very clear.

If it's about e.g. a missing column, that should be what's expressed in the name, rather than "different schema".

Reply:
Naming is confusing indeed. I'll update soon.

Reply:
> You use different schema terminology a lot here. Can you explain what you mean by it

I prepend `DifferentSchema` to the classes making use of different table names and columns(in some cases). See all the classes below:
<img width="281" alt="image" src="https://user-images.githubusercontent.com/54532330/195593292-96a71d75-99cc-43c3-bab9-9f3841b5e8f8.png">

More natural name for `ResourceTenantForDifferentSchema` can be just `ResourceTenant` but same class exists already. So this name feels appropriate.

Reply:
> I prepend DifferentSchema

You mean suffix, right?

> making use of different table names and columns

So essentially a case where the tenant DB has less/more columns than the central DB?

Some thoughts:
1. It can indeed be difficult to think of a better term than "different schema", so the current term paired with a clear comment could work
2. However: **in what way** is it different? Does it have less or more columns than the other model?
3. And what's the other model here — the central schema? Since you're using "different schema": different to what? Is the schema of `users` completely identical for the central DB and the tenant DBs in the `master` version of these tests? So your point here is that *only in these tests* we introduce a different schema for the `users` table in the tenant DB, compared to the central DB.

Reply:
> You mean suffix, right?

yes, I meant suffix. 🤦‍♂️

> 2. However: in what way is it different? Does it have fewer or more columns than the other model?

1) both tables have different names. 2) Columns are actually the same in both models but different from the master version or first setup for sync resources (ResourceUser, CentralUser). 

> 2. Is the schema of users completely identical for the central DB and the tenant DBs in the master version of these tests?

No. They have the same table names, but the columns are somewhat different. I mean, that was the point of an original issue when one model has an extra column, it causes a problem. We are trying to solve it by providing list of attributes. 
At some point, I thought having a different setup like this was repetitive. We are kind of testing the same thing in the master version, except the table names are the same.

Reply:
> No. They have the same table names, but the columns are somewhat different

Even **in the `master` version of these tests?** Meaning, the tests that we already have in the `master` branch in this file.